cfme/tests/cloud/test_providers.py

```diff
@@ -239,7 +239,7 @@ def test_cloud_provider_add_with_bad_credentials(provider, enable_regions):
         flash = 'Credential validation was not successful: Invalid Google JSON key'
         default_credentials.service_account = '{"test": "bad"}'
     elif provider.one_of(OpenStackProvider):
-        for endp_name in provider.endpoints.keys():
+        for endp_name in list(provider.endpoints.keys()):
```

Comment:
I think conversion not needed for iteration.

Reply:
Its not because of iteration, its because `provider.endpoints` is modified within the loop, changing the length of the iterator resulting in a RuntimeError.

Wrapping this in a list explicitly means the for loop traverses the list object, not the keys iterator. This list doesn't change size when `provider.endpoints` is modified.

Thanks @izapolsk ;)